### PR TITLE
Fix some bugs with find

### PR DESCRIPTION
**Pull request by w-yuchen**
There were some unhandled exceptions previously.

### CODE REVIEW

File: src/main/java/seedu/address/model/property/PropertyAddressPredicate.java [hunk 1]
```diff
@@ -13,8 +13,11 @@
     /**
      * Constructs a PropertyAddressPredicate.
      */
-    public PropertyAddressPredicate(String address) {
+    public PropertyAddressPredicate(String address) throws IllegalArgumentException {
         requireNonNull(address);
+        if (address.trim().isEmpty()) {
+            throw new IllegalArgumentException("Address cannot be empty. ");
```
Comment (w2vgd) [inline src/main/java/seedu/address/model/property/PropertyAddressPredicate.java]: Maybe change to `"Address given is empty"` to make error message consistent throughout?

File: src/main/java/seedu/address/model/property/PropertyNamePredicate.java [hunk 2]
```diff
@@ -14,10 +14,14 @@
...
     /**
      * Constructs a {@code PropertyNamePredicate}
+     * @throws IllegalArgumentException when keyword is empty.
      */
-    public PropertyNamePredicate(List<String> names) {
+    public PropertyNamePredicate(List<String> names) throws IllegalArgumentException {
         this.keywords = new ArrayList<>();
         for (String s : names) {
+            if (s.trim().isEmpty()) {
+                throw new IllegalArgumentException("Name cannot be empty. ");
```
Comment (w2vgd) [inline src/main/java/seedu/address/model/property/PropertyNamePredicate.java]: Same with previous comment, try to make the error message consistent

File: src/main/java/seedu/address/model/property/PropertyRemarksPredicate.java [hunk 3]
```diff
@@ -8,13 +8,20 @@
 public class PropertyRemarksPredicate implements Predicate<Property> {
     private final String keywords;
 
+    /**
+     * Constructs a {@code PropertyRemarksPredicate} with a keyword.
+     * @throws IllegalArgumentException when keyword is empty.
+     */
     public PropertyRemarksPredicate(String keywords) throws IllegalArgumentException {
+        if (keywords.trim().isEmpty()) {
+            throw new IllegalArgumentException("Remark keywords cannot be empty. ");
```
Comment (w2vgd) [inline src/main/java/seedu/address/model/property/PropertyRemarksPredicate.java]: Same as previous

**Review by w2vgd — COMMENTED** [review]
Some comments

File: src/main/java/seedu/address/model/property/PropertyAddressPredicate.java [hunk 4]
```diff
@@ -13,8 +13,11 @@
     /**
      * Constructs a PropertyAddressPredicate.
      */
-    public PropertyAddressPredicate(String address) {
+    public PropertyAddressPredicate(String address) throws IllegalArgumentException {
         requireNonNull(address);
+        if (address.trim().isEmpty()) {
+            throw new IllegalArgumentException("Address give is empty. ");
```
Comment (w2vgd) [inline src/main/java/seedu/address/model/property/PropertyAddressPredicate.java]: Small typo `give` instead of `given`

**Review by w2vgd — COMMENTED** [review]
LGTM with a small typo

File: src/main/java/seedu/address/model/property/PropertyAddressPredicate.java (hunk 4 above)
Comment (w-yuchen) [reply src/main/java/seedu/address/model/property/PropertyAddressPredicate.java]: fixed

**Review by dvdweien — APPROVED** [review]
LGTM

**Review by w2vgd — APPROVED** [review]
LGTM

**Pull request merged**